### PR TITLE
PLAT-76029: Improve performance of ilib string mapping

### PR DESCRIPTION
**Pull request by ryanjduffy**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
The construction cost of a ResBundle is small (~1ms at 6x slowdown) but is significant relative to the overall cost of formatting a message (~0.1ms at 6x slowdown).

**Timing for formatting 5 messages**
image

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
* Update the ilib formatter to construct a "reverse" map of translated string to source strings per locale
* Memoize the map retrieval by locale which changes infrequently

The result is a reduction of the ilib formatting to ~0.01ms at 6x slowdown.

**Timing for formatting 5 messages**
image

### CODE REVIEW

File: preset/ilib.js [hunk 1]
```diff
@@ -1,33 +1,55 @@
...
-			// different case during render.
-			const upperValue = msg[msgKey].toUpperCase();
-			const value = Object.keys(map).find(key => upperValue === map[key].toUpperCase());
-
-			if (value) {
-				msg[msgKey] = value;
-			}
-		});
+		Object.keys(msg)
+			.filter(key => key !== 'type' && typeof msg[key] === 'string')
```
Comment (webOS101) [inline preset/ilib.js:30]: Is `'time'` not a filtered key any more?  If not, should we update the comment above?  Or, is it that it's a non-string type and now we're more strict?  Either way, I think something needs to be clarified.

Comment (ryanjduffy) [reply preset/ilib.js:30]: It's filtered by its `typeof` but i'll update the comment to reflect that.

Comment (webOS101) [reply preset/ilib.js:30]: Is this a critical path?  Would it make sense to dispense with the `.filter()` step and do it inline with the `forEach()`?  Too much micro-opt?

Comment (ryanjduffy) [reply preset/ilib.js:30]: Seems like a micro opt to me. In my 3 key test, this method is only taking 0.01ms. Assuming it's linear, there'd have to be a lot of keys in the message payload to make a meaningful impact.

Comment (webOS101) [reply preset/ilib.js:30]: My only thought was along the lines of reducing pressure on GC.  But, as you say, it's probably not likely they are large arrays being allocated/deallocated.

**Review by JayCanuck — APPROVED** [review]
:LGTM:

**Pull request merged**